### PR TITLE
HeartbeatV2 improvements and fixes

### PR DESCRIPTION
**Pull request by sstanculeanu**
#### Changes
- renamed `ShardValidatorInfo` to `DirectConnectionInfo` as it collides with other components
- made ShardId string as due to proto3 shardId 0 is not accepted
- added `dataPacker` on `peerAuthenticationResolver`
- fixed few bugs and added extra logs
- fixed the topic for heartbeat v2 sender and whitelisted topics on antiflood handler

### CODE REVIEW

File: dataRetriever/resolvers/peerAuthenticationResolver.go [hunk 1]
```diff
@@ -241,27 +250,24 @@ func (res *peerAuthenticationResolver) resolveMultipleHashesRequest(hashesBuff [
...
-
-	return res.sendData(dataBuff, hashesBuff, 0, 0, pid)
-}
-
-// sendLargeDataBuff splits dataBuff into chunks and sends a message for the first chunk
-func (res *peerAuthenticationResolver) sendLargeDataBuff(dataBuff [][]byte, reference []byte, chunkSize int, pid core.PeerID) error {
-	maxChunks := res.getMaxChunks(dataBuff)
-	chunk, err := res.extractChunk(dataBuff, 0, chunkSize, maxChunks)
+func (res *peerAuthenticationResolver) sendPeerAuthsForHashes(dataBuff [][]byte, pid core.PeerID) error {
+	buffsToSend, err := res.dataPacker.PackDataInChunks(dataBuff, maxBuffToSendPeerAuthentications)
```
Comment (iulianpascalau) [inline dataRetriever/resolvers/peerAuthenticationResolver.go:258]: :+1:

File: factory/heartbeatV2Components.go [hunk 2]
```diff
@@ -114,13 +114,16 @@ func (hcf *heartbeatV2ComponentsFactory) Create() (*heartbeatV2Components, error
...
+	heartbeatTopic := common.HeartbeatV2Topic + shardC.CommunicationIdentifier(shardC.SelfId())
+
 	cfg := hcf.config.HeartbeatV2
 
 	argsSender := sender.ArgSender{
 		Messenger:                          hcf.networkComponents.NetworkMessenger(),
 		Marshaller:                         hcf.coreComponents.InternalMarshalizer(),
 		PeerAuthenticationTopic:            common.PeerAuthenticationTopic,
-		HeartbeatTopic:                     common.HeartbeatV2Topic,
+		HeartbeatTopic:                     heartbeatTopic,
```
Comment (iulianpascalau) [inline factory/heartbeatV2Components.go:126]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: process/interceptors/processor/peerAuthenticationInterceptorProcessor.go [hunk 3]
```diff
@@ -82,18 +82,18 @@ func (paip *peerAuthenticationInterceptorProcessor) Save(data process.Intercepte
 		return err
 	}
 
-	paip.peerAuthenticationCacher.Put(fromConnectedPeer.Bytes(), interceptedPeerAuthenticationData.Message(), interceptedPeerAuthenticationData.SizeInBytes())
-
-	return paip.updatePeerInfo(interceptedPeerAuthenticationData.Message())
+	return paip.updatePeerInfo(interceptedPeerAuthenticationData.Message(), interceptedPeerAuthenticationData.SizeInBytes())
```
Comment (iulianpascalau) [inline process/interceptors/processor/peerAuthenticationInterceptorProcessor.go:85]: :+1:

**Review by iulianpascalau — APPROVED** [review]

File: heartbeat/monitor/monitor.go [hunk 4]
```diff
@@ -125,7 +125,7 @@ func (monitor *heartbeatV2Monitor) GetHeartbeats() []data.PubKeyHeartbeat {
 func (monitor *heartbeatV2Monitor) parseMessage(pid core.PeerID, message interface{}, numInstances map[string]uint64) (data.PubKeyHeartbeat, error) {
 	pubKeyHeartbeat := data.PubKeyHeartbeat{}
 
-	heartbeatV2, ok := message.(heartbeat.HeartbeatV2)
+	heartbeatV2, ok := message.(*heartbeat.HeartbeatV2)
```
Comment (iulianpascalau) [inline heartbeat/monitor/monitor.go:128]: :1st_place_medal:

**Review by iulianpascalau — DISMISSED** [review]

File: process/interceptors/processor/directConnectionInfoInterceptorProcessor.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+type shardProvider interface {
+	ShardID() string
+}
+
+// ArgDirectConnectionInfoInterceptorProcessor is the argument for the interceptor processor used for direct connection info
+type ArgDirectConnectionInfoInterceptorProcessor struct {
+	PeerShardMapper process.PeerShardMapper
+}
+
+type DirectConnectionInfoInterceptorProcessor struct {
```
Comment (ssd04) [inline process/interceptors/processor/directConnectionInfoInterceptorProcessor.go:20]: i think this can be set to be unexported

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**